src/Microsoft.ApiDesignGuidelines.Analyzers/Core/ImplementStandardExceptionConstructors.cs

```diff
@@ -27,23 +31,96 @@ public abstract class ImplementStandardExceptionConstructorsAnalyzer : Diagnosti
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
-                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
-        internal static DiagnosticDescriptor AccessibilityRule = new DiagnosticDescriptor(RuleId,
-                                                                             s_localizableTitle,
-                                                                             s_localizableMessageAccessibility,
-                                                                             DiagnosticCategory.Design,
-                                                                             DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
-                                                                             description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182151.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(MissingConstructorRule, AccessibilityRule);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(MissingConstructorRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            // Analyze named types 
+            analysisContext.RegisterSymbolAction(symbolContext =>
+            {
+                AnalyzeSymbol(symbolContext.Symbol, symbolContext);
+
+            }, SymbolKind.NamedType);
+        }
+
+        //abstract methods, which the language specific analyzers implements - these will return the required constructor method signatures for CSharp/Basic
+        protected abstract string GetParameterLessConstructorMessage(ISymbol symbol);
+        protected abstract string GetConstructorWithStringTypeParameter(ISymbol symbol);
+        protected abstract string GetConstructorWithStringAndExceptionTypeParameter(ISymbol symbol);
+
+        private void AnalyzeSymbol(ISymbol symbol, SymbolAnalysisContext context)
+        {
+            var namedTypeSymbol = context.Symbol as INamedTypeSymbol;
+
+            //Check if type derives from Exception type
+            if (namedTypeSymbol.BaseType.MetadataName == context.Compilation.GetTypeByMetadataName("System.Exception").MetadataName)
```

Comment:
Instead of fetching the exception type on every call to AnalyzeSymbol, you could register a CompilationStart action, fetch the exception type and pass that down to the Symbol action.